packages/nuxt/src/imports/module.ts

```diff
@@ -176,7 +180,7 @@ function addDeclarationTemplates (ctx: Unimport, options: Partial<ImportsOptions
   addTemplate({
     filename: 'types/imports.d.ts',
     getContents: async () => {
-      const imports = await ctx.getImports().then(r => r.filter(i => !i.type))
+      const imports = await ctx.getImports()
```

Comment:
/cc @danielroe do you remember why you added this filter in https://github.com/nuxt/nuxt/pull/22476?

Reply:
No 🤔

Perhaps I was thinking we didn't need to _cache_ the type-only definitions as they wouldn't be emitted here...?

I think we can remove it safely.